controllers/nodenetworkconfigurationpolicy_controller_test.go

```diff
@@ -62,4 +71,115 @@ var _ = Describe("NodeNetworkConfigurationPolicy controller predicates", func()
 				ReconcileUpdate: true,
 			}),
 	)
+
+	type incrementUnavailableNodeCountCase struct {
+		currentUnavailableNodeCount  int
+		expectedUnavailableNodeCount int
+		previousEnactmentConditions  func(*shared.ConditionList, string)
+		shouldIncrement              bool
+		shouldConflict               bool
+	}
+	DescribeTable("when claimNodeRunningUpdate is called and",
+		func(c incrementUnavailableNodeCountCase) {
+			reconciler := NodeNetworkConfigurationPolicyReconciler{}
+			s := scheme.Scheme
+			s.AddKnownTypes(nmstatev1beta1.GroupVersion,
+				&nmstatev1beta1.NodeNetworkConfigurationPolicy{},
+				&nmstatev1beta1.NodeNetworkConfigurationEnactment{},
+				&nmstatev1beta1.NodeNetworkConfigurationEnactmentList{},
+			)
+
+			nncp := nmstatev1beta1.NodeNetworkConfigurationPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+				},
+				Status: shared.NodeNetworkConfigurationPolicyStatus{
+					UnavailableNodeCount: c.currentUnavailableNodeCount,
+				},
+			}
+			nnce := nmstatev1beta1.NodeNetworkConfigurationEnactment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: shared.EnactmentKey(nodeName, nncp.Name).Name,
+				},
+				Status: shared.NodeNetworkConfigurationEnactmentStatus{},
+			}
+
+			// simulate NNCE existnce/non-existence by setting conditions
+			c.previousEnactmentConditions(&nnce.Status.Conditions, "")
+
+			objs := []runtime.Object{&nncp, &nnce}
+
+			// Create a fake client to mock API calls.
+			clb := fake.ClientBuilder{}
+			clb.WithScheme(s)
+			clb.WithRuntimeObjects(objs...)
+			cl := clb.Build()
+
+			reconciler.Client = cl
+			reconciler.Log = ctrl.Log.WithName("controllers").WithName("NodeNetworkConfigurationPolicy")
+
+			shouldIncrement := reconciler.shouldIncrementUnavailableNodeCount(&nnce.Status.Conditions)
+			Expect(shouldIncrement).To(Equal(c.shouldIncrement))
+
+			if shouldIncrement {
+				err := reconciler.incrementUnavailableNodeCount(&nncp)
+				if c.shouldConflict {
+					Expect(err).Should(WithTransform(apierrors.IsConflict, BeTrue()), "should conflict")
+				} else {
+					Expect(err).ToNot(HaveOccurred())
+				}
+			}
```

Comment:
Can we exercise this code using the Reconcile function instead of the helpers ? If after a refactor one of the helpers is not being called at Reconcile this test will be not converting the issue.